src/wasm2asm.h

```diff
@@ -950,36 +976,60 @@ Ref Wasm2AsmBuilder::processFunctionBody(Function* func, IString result) {
+            case CtzInt32:
+              return ValueBuilder::makeCall(MATH_CTZ32, value);
+            case PopcntInt32:
+              return ValueBuilder::makeCall(MATH_POPCNT32, value);
             default: abort();
           }
         }
         case f32:
         case f64: {
           Ref ret;
           switch (curr->op) {
-            case Neg:           ret = ValueBuilder::makeUnary(MINUS, value); break;
-            case Abs:           ret = ValueBuilder::makeCall(MATH_ABS, value); break;
-            case Ceil:          ret = ValueBuilder::makeCall(MATH_CEIL, value); break;
-            case Floor:         ret = ValueBuilder::makeCall(MATH_FLOOR, value); break;
-            case Trunc:         ret = ValueBuilder::makeCall(MATH_TRUNC, value); break;
-            case Nearest:       ret = ValueBuilder::makeCall(MATH_NEAREST, value); break;
-            case Sqrt:          ret = ValueBuilder::makeCall(MATH_SQRT, value); break;
-            //case TruncSFloat32: ret = ValueBuilder::makePrefix(B_NOT, ValueBuilder::makePrefix(B_NOT, value)); break;
-            case PromoteFloat32:
-            //case ConvertSInt32: ret = ValueBuilder::makePrefix(PLUS, ValueBuilder::makeBinary(value, OR, ValueBuilder::makeNum(0))); break;
-            //case ConvertUInt32: ret = ValueBuilder::makePrefix(PLUS, ValueBuilder::makeBinary(value, TRSHIFT, ValueBuilder::makeNum(0))); break;
-            case DemoteFloat64: ret = value; break;
-            default: std::cerr << curr << '\n'; abort();
+            case NegFloat32:
+            case NegFloat64:
+              ret = ValueBuilder::makeUnary(MINUS, value);
+              break;
+            case AbsFloat32:
+            case AbsFloat64:
+              ret = ValueBuilder::makeCall(MATH_ABS, value);
+              break;
+            case CeilFloat32:
+            case CeilFloat64:
+              ret = ValueBuilder::makeCall(MATH_CEIL, value);
+              break;
+            case FloorFloat32:
+            case FloorFloat64:
+              ret = ValueBuilder::makeCall(MATH_FLOOR, value);
+              break;
+            case TruncFloat32:
+            case TruncFloat64:
+              ret = ValueBuilder::makeCall(MATH_TRUNC, value);
+              break;
+            case NearestFloat32:
+            case NearestFloat64:
+              ret = ValueBuilder::makeCall(MATH_NEAREST, value);
+              break;
+            case SqrtFloat32:
+            case SqrtFloat64:
+              ret = ValueBuilder::makeCall(MATH_SQRT, value);
+              break;
+            // TODO: more complex unary conversions
+            default:
+              std::cerr << curr << '\n';
+              abort();
           }
           if (curr->type == f32) { // doubles need much less coercing
             return makeAsmCoercion(ret, ASM_FLOAT);
           }
           return ret;
         }
-        default: abort();
+        default:
+          std::cerr << curr << '\n';
```

Comment:
Yay error messages.

Might be worth adding more context though? "Unexpected type while visiting const" or something?